src/richchk/model/richchk/rich_chk_section.py

```diff
@@ -27,8 +25,5 @@ def cast(cls, rich_chk_section: "RichChkSection", chk_section_type: Type[_T]) ->
         :param chk_section_type:
         :return:
         """
-        # we only want non-abstract fields, dataclass_wizard seems to pick these up
-        as_dict = dataclasses.asdict(rich_chk_section)
-        section = dataclass_wizard.fromdict(chk_section_type, as_dict)
-        assert isinstance(section, chk_section_type)
-        return section
```

Comment:
it doesn't seem necessary to re-create the object.  Just assert it's type to keep type checker happy.